src/Preview.js

```diff
@@ -36,15 +39,57 @@ const MarkdownPreview = ({ path }) => {
     ['file', { user, repo, branch, path }],
     getFileContent
   )
-  const { status, data: renderedMarkdown } = useQuery(
-    text && ['markdown', { user, repo, text }],
-    getMarkdown
-  )
 
-  if (status === 'loading') {
-    return <LoadingComponent />
+  const staticUrlRegex = new RegExp(/^https?:/)
+
+  const transformImageUri = uri => {
+    if (!staticUrlRegex.test(uri)) {
+      return `https://github.com/${user}/${repo}/raw/${branch}/${uri}`
+    } else {
+      return uri.replace('/blob/', '/raw/').replace('http://', 'https://')
+    }
```

Comment:
I love being able to walk to markdown AST and correcting the URLs!

What I'm concerned about though, is forgoing the github markdown redering endpoint and doing all of the complicated work ourselves. Github-flavored-markdown has become pretty complex, and we'd need to handle _all_ of the things it does if we decide to render it ourselves (code highlighting, commit-, branch-, user-, pr-, issue references formatting and linking etc.).

I'm unwilling to take on all that complexity just to fix some links. But! I figure we could have the cake and eat it too :)

I propose that we process and fix the links _before_ we send it to the GH markdown rendering service. That takes care of 100% of the complex stuff that GH is doing with GFM and we get to fix our srcs and hrefs too. What do you think?

Reply:
I totally agree with you, I didn't think about all the other magical unknown stuff GFM has and does. I'll update this PR in the next few days when I find the time to do so.